core/src/main/java/com/linecorp/armeria/server/metric/PrometheusExpositionService.java

```diff
@@ -85,11 +89,30 @@ public PrometheusExpositionService(CollectorRegistry collectorRegistry) {
 
     @Override
     protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) throws Exception {
-        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
-        try (OutputStreamWriter writer = new OutputStreamWriter(stream)) {
-            TextFormat.write004(writer, collectorRegistry.metricFamilySamples());
-        }
-        return HttpResponse.of(HttpStatus.OK, CONTENT_TYPE_004, stream.toByteArray());
+        final HttpResponseWriter responseWriter = HttpResponse.streaming();
+        responseWriter.write(ResponseHeaders.builder(HttpStatus.OK).contentType(CONTENT_TYPE_004).build());
+        ctx.blockingTaskExecutor().execute(() -> {
+            try (BufferedWriter writer = new BufferedWriter(new Writer() {
```

Comment:
Curious if we need to adjust buffer size

Reply:
Oops, I just saw this. I left my opinion here. https://github.com/line/armeria/pull/3929#discussion_r749087882
:-)